Games/RomMInstallController.cs

```diff
@@ -96,10 +96,30 @@ public override void Install(InstallActionArgs args)
                             ExtractNestedArchives(installDir);
                         }
 
+                        List<string> supportedFileTypes = null;
+                        if (info.Mapping.EmulatorProfile is CustomEmulatorProfile)
+                        {
+                            var customProfile = info.Mapping.EmulatorProfile as CustomEmulatorProfile;
+                            supportedFileTypes = customProfile.ImageExtensions;
+                        }
+                        else if (info.Mapping.EmulatorProfile is BuiltInEmulatorProfile)
+                        {
+                            var builtInProfile = (info.Mapping.EmulatorProfile as BuiltInEmulatorProfile);
+                            supportedFileTypes = API.Instance.Emulation.Emulators
+                                .FirstOrDefault(e => e.Id == info.Mapping.Emulator.BuiltInConfigId)?
+                                .Profiles
+                                .FirstOrDefault(p => p.Name == builtInProfile.Name)?
+                                .ImageExtensions;
+                        }
+
+                        var searchPatterns = (supportedFileTypes != null || supportedFileTypes.Count == 0)
+                            ? new List<string> { "*" }
+                            : supportedFileTypes.Select(x => "*." + x);
+
                         // Add just the ROM files to the list
-                        string[] actualRomFiles = Directory.GetFiles(installDir, "*", SearchOption.AllDirectories)
-                            .Where(file => !file.EndsWith(".m3u", StringComparison.OrdinalIgnoreCase))
-                            .ToArray();
+                        string[] actualRomFiles = searchPatterns.SelectMany(
+                            searchPattern => Directory.GetFiles(installDir, searchPattern, SearchOption.AllDirectories)
```

Comment:
i think we still need to ignore `m3u` files in the case where they're in the ZIP

Reply:
That won't interfere with my use case, but I'm curious: won't it look like a bug if it ignores `m3u` files when a user has specified them as supported files in the emulator configuration?

Reply:
yes in the case where the user has specified it, but in the one where `supportedFileTypes == null || supportedFileTypes.Count == 0` the search pattern `new List<string> { "*" }` should still ignore `.m3u` files to match the current behaviour.

Reply:
Ah, got it! I will work on that